champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/ranger/passives/HeavyArrows.java

```diff
@@ -0,0 +1,115 @@
+import me.mykindos.betterpvp.champions.champions.skills.Skill;
+import me.mykindos.betterpvp.champions.champions.skills.types.EnergySkill;
+import me.mykindos.betterpvp.champions.champions.skills.types.PassiveSkill;
+import me.mykindos.betterpvp.core.components.champions.Role;
+import me.mykindos.betterpvp.core.components.champions.SkillType;
+import me.mykindos.betterpvp.core.components.champions.events.PlayerCanUseSkillEvent;
+import me.mykindos.betterpvp.core.framework.updater.UpdateEvent;
+import me.mykindos.betterpvp.core.listener.BPvPListener;
+import me.mykindos.betterpvp.core.utilities.UtilServer;
+import org.bukkit.Location;
+import org.bukkit.Particle;
+import org.bukkit.entity.Arrow;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.entity.EntityShootBowEvent;
+import org.bukkit.util.Vector;
+
+import java.util.Iterator;
+import java.util.WeakHashMap;
+
+@Singleton
+@BPvPListener
+public class HeavyArrows extends Skill implements PassiveSkill, EnergySkill{
+
+    private final WeakHashMap<Arrow, Location> arrows = new WeakHashMap<>();
+
+    public double basePushBack;
+
+    @Inject
+    public HeavyArrows(Champions champions, ChampionsManager championsManager) {
+        super(champions, championsManager);
+    }
+
+    @Override
+    public String getName() {
+        return "Heavy Arrows";
+    }
+
+    @Override
+    public String[] getDescription(int level) {
+        return new String[]{
+                "The arrows you shoot are heavy",
+                "",
+                "For every arrow you shoot you will be",
+                "pushed backwards (unless crouching)",
+                "",
+                "Energy used per shot: <val>"+energy
+        };
+    }
+
+    @Override
+    public Role getClassType() {
+        return Role.RANGER;
+    }
+
+    @UpdateEvent
+    public void update() {
+        Iterator<Arrow> it = arrows.keySet().iterator();
+        while (it.hasNext()) {
+            Arrow next = it.next();
+            if (next == null || next.isDead()) {
+                it.remove();
+            } else {
+                Location location = next.getLocation().add(new Vector(0, 0.25, 0));
+                Particle.CRIT_MAGIC.builder().location(location).receivers(60).extra(0).spawn();
+            }
+        }
+    }
+
+    @EventHandler
+    public void onShoot(EntityShootBowEvent event) {
+        if (!(event.getEntity() instanceof Player player)) return;
+        if (!(event.getProjectile() instanceof Arrow arrow)) return;
+
+        int level = getLevel(player);
+        if (level > 0) {
+            PlayerCanUseSkillEvent skillEvent = UtilServer.callEvent(new PlayerCanUseSkillEvent(player, this));
+            if (!skillEvent.isCancelled()) {
+                arrows.put(arrow, arrow.getLocation());
+
+                if (!player.isSneaking()) {
+
+                    if (championsManager.getEnergy().use(player, getName(), getEnergy(level) / 2, false)) {
```

Comment:
What's the need to half the energy consumption here? Wouldn't this make it inconsistent with the number specified in the config

Reply:
to be honest I have no idea what that if statement does I just copied it from another class and it works

Reply:
`EnergyHandler#use(Player, String, double, boolean)` either consumes the energy you passed in (the double argument) and returns true if the player had enough energy, or returns false if the player didn't have enough energy. Here you're trying to consume x amount of energy (`getEnergy(level)`) but you're halving it, effectively making it so they only consume half the energy they're meant to. You'd just need to remove the / 2.

```suggestion
                    if (championsManager.getEnergy().use(player, getName(), getEnergy(level), false)) {
```